### PR TITLE
Update sshd crypto policy for CC

### PR DESCRIPTION
**Pull request by vojtapolasek**
#### Description:

Add supported public key algorithms to sshd crypto-policy - rsa-sha2-256,rsa-sha2-512

#### Rationale:

Update supported algorithms for common criteria

### CODE REVIEW

File: linux_os/guide/system/software/integrity/crypto/harden_sshd_crypto_policy/bash/shared.sh [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # platform = Oracle Linux 8,Red Hat Enterprise Linux 8,Red Hat Virtualization 4,multi_platform_fedora
 
 
-cp="CRYPTO_POLICY='-oCiphers=aes128-ctr,aes256-ctr,aes128-cbc,aes256-cbc -oMACs=hmac-sha2-256,hmac-sha2-512 -oGSSAPIKeyExchange=no -oKexAlgorithms=diffie-hellman-group14-sha1,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521 -oHostKeyAlgorithms=ssh-rsa,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384 -oPubkeyAcceptedKeyTypes=ssh-rsa,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384'"
+cp="CRYPTO_POLICY='-oCiphers=aes128-ctr,aes256-ctr,aes128-cbc,aes256-cbc -oMACs=hmac-sha2-256,hmac-sha2-512 -oGSSAPIKeyExchange=no -oKexAlgorithms=diffie-hellman-group14-sha1,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521 -oHostKeyAlgorithms=ssh-rsa,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384 -oPubkeyAcceptedKeyTypes=ssh-rsa,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,rsa-sha2-256,rsa-sha2-512'"
```
Comment (yuumasato) [inline linux_os/guide/system/software/integrity/crypto/harden_sshd_crypto_policy/bash/shared.sh:4]: LGTM, but..
@jjaburek @stevegrubb Can the new algorithms just be appended? Should they be in inserted in a specific position?

**Comment by stevegrubb** [comment]
In a lot of cases, sshd/ssh uses the order given as the preferred order. So, generally you want things ordered from strongest to weakest. I don't think this particular setting matters on the server side. But it might on the client side. The safest thing to do is just go from strong to weak.

**Comment by mildas** [comment]
**Changes identified:**
Rule harden_sshd_crypto_policy:
 Template usage changed in OVAL check.
 Found change in bash remediation.

**Recommended tests to execute:**
 build_product ocp4
 test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate bash --datastream build/ssg-ocp4-ds.xml harden_sshd_crypto_policy

**Comment by vojtapolasek** [comment]
> In a lot of cases, sshd/ssh uses the order given as the preferred order. So, generally you want things ordered from strongest to weakest. I don't think this particular setting matters on the server side. But it might on the client side. The safest thing to do is just go from strong to weak.

So dou want me to also sort ciphers etc? Cause now they are afaik not sorted from strongest to weakest. How about this?
-oCiphers=aes256-ctr,aes128-ctr,aes256-cbc,aes128-cbc -oMACs=hmac-sha2-512,hmac-sha2-256 -oGSSAPIKeyExchange=no -oKexAlgorithms=ecdh-sha2-nistp521,ecdh-sha2-nistp384,ecdh-sha2-nistp256,diffie-hellman-group14-sha1 -oHostKeyAlgorithms=ecdsa-sha2-nistp384,ecdsa-sha2-nistp256,ssh-rsa -oPubkeyAcceptedKeyTypes=ecdsa-sha2-nistp384,ecdsa-sha2-nistp256,rsa-sha2-512,rsa-sha2-256,ssh-rsa

**Comment by mildas** [comment]
**Changes identified:**
Rule harden_sshd_crypto_policy:
 Template usage changed in OVAL check.
 Found change in bash remediation.

**Recommended tests to execute:**
 build_product ocp4
 test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate bash --datastream build/ssg-ocp4-ds.xml harden_sshd_crypto_policy

File: linux_os/guide/system/software/integrity/crypto/harden_sshd_crypto_policy/bash/shared.sh [hunk 2]
```diff
@@ -1,7 +1,6 @@
 # platform = Oracle Linux 8,Red Hat Enterprise Linux 8,Red Hat Virtualization 4,multi_platform_fedora
 
-
-cp="CRYPTO_POLICY='-oCiphers=aes128-ctr,aes256-ctr,aes128-cbc,aes256-cbc -oMACs=hmac-sha2-256,hmac-sha2-512 -oGSSAPIKeyExchange=no -oKexAlgorithms=diffie-hellman-group14-sha1,ecdh-sha2-nistp256,ecdh-sha2-nistp384,ecdh-sha2-nistp521 -oHostKeyAlgorithms=ssh-rsa,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384 -oPubkeyAcceptedKeyTypes=ssh-rsa,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384'"
+cp="CRYPTO_POLICY='-oCiphers=aes256-ctr,aes128-ctr,aes256-cbc,aes128-cbc -oMACs=hmac-sha2-512,hmac-sha2-256 -oGSSAPIKeyExchange=no -oKexAlgorithms=ecdh-sha2-nistp521,ecdh-sha2-nistp384,ecdh-sha2-nistp256,diffie-hellman-group14-sha1 -oHostKeyAlgorithms=ssh-rsa,ecdsa-sha2-nistp384,ecdsa-sha2-nistp256 -oPubkeyAcceptedKeyTypes=rsa-sha2-512,rsa-sha2-256,ssh-rsa,ecdsa-sha2-nistp384,ecdsa-sha2-nistp256'"
```
Comment (yuumasato) [inline linux_os/guide/system/software/integrity/crypto/harden_sshd_crypto_policy/bash/shared.sh:3]: @t8m Hi, does the order of the algorithms for opensshserver matter?

If so, can you help review order of the algorithms?

Comment (t8m) [reply linux_os/guide/system/software/integrity/crypto/harden_sshd_crypto_policy/bash/shared.sh:3]: Yes, it does. I think the new order is OK.

Comment (yuumasato) [reply linux_os/guide/system/software/integrity/crypto/harden_sshd_crypto_policy/bash/shared.sh:3]: Awesome, thank you for the review.

**Pull request merged**